analytics_test.go

```diff
@@ -156,7 +157,9 @@ func mockServer() (chan []byte, *httptest.Server) {
 
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		buf := bytes.NewBuffer(nil)
-		io.Copy(buf, r.Body)
+		reader, _ := gzip.NewReader(r.Body)
```

Comment:
`mockServer` should also support verifying the expected payload received during tests. Tests should fail an unexpected request payload is received.

Reply:
> should also support verifying the expected payload received during tests.

Didn't get it quitely.

Reply:
`mockServer` returns both a channel which contains the body as received by the server and a test server.
But we never verify the received body in any of our test cases. This alows for bugs like the one we observed during the lifetime of this PR (client was sending an empty gzipped payload) to happen, whereas it would be really easy to guard against similar issues. We just need to verify in our tests that the received payload is the expected one, i.e. what we sent. Is this something you could include in the context of this pr?

Reply:
Got it. We need to check all the test cases. We can't include this atm. I am creating a task for this.